nvflare/app_common/ccwf/ccwf_job.py

```diff
@@ -0,0 +1,308 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Optional
+
+from nvflare.apis.executor import Executor
+from nvflare.app_common.abstract.aggregator import Aggregator
+from nvflare.app_common.abstract.metric_comparator import MetricComparator
+from nvflare.app_common.abstract.model_persistor import ModelPersistor
+from nvflare.app_common.abstract.shareable_generator import ShareableGenerator
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.ccwf.common import Constant, CyclicOrder
+from nvflare.fuel.utils.validation_utils import check_object_type
+from nvflare.job_config.api import ControllerApp, ExecutorApp, FedJob
+from nvflare.widgets.widget import Widget
+
+from .cse_client_ctl import CrossSiteEvalClientController
+from .cse_server_ctl import CrossSiteEvalServerController
+from .cyclic_client_ctl import CyclicClientController
+from .cyclic_server_ctl import CyclicServerController
+from .swarm_client_ctl import SwarmClientController
+from .swarm_server_ctl import SwarmServerController
+
+
+class FedAvgServerConfig:
+    def __init__(
+        self,
+        num_rounds: int,
+        start_round: int = 0,
+        start_task_timeout=Constant.START_TASK_TIMEOUT,
+        configure_task_timeout=Constant.CONFIG_TASK_TIMEOUT,
+        participating_clients=None,
+        result_clients=None,
+        starting_client: str = "",
+        max_status_report_interval: float = Constant.PER_CLIENT_STATUS_REPORT_TIMEOUT,
+        progress_timeout: float = Constant.WORKFLOW_PROGRESS_TIMEOUT,
+        private_p2p: bool = True,
+        aggr_clients=None,
+        train_clients=None,
+    ):
```

Comment:
since this object is user facing, could we add docstring to it and explain the arguments?